infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/AddressPointsCache.kt

```diff
@@ -28,6 +37,8 @@ class AddressPointsCache(
     val geocodingCacheService: GeocodingCacheService,
 ) {
     private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+    private val cache: Cache<AddressPointCacheKey, AlignmentAddresses?> =
+        Caffeine.newBuilder().maximumSize(ADDRESS_POINT_CACHE_SIZE).expireAfterAccess(layoutCacheDuration).build()
```

Comment:
Mites meillä on muualla hoidettu cachejen expiraus? Yleensähän aika on meillä aika huono cacheputsari, kun data muuttuu harvakseltaan (mutta muutoksen jälkeen kaikki pitäisi invalidoida heti)

Reply:
Tuo duration on tunnin. Sinä aikana tietenkin n tuoreinta pyyntöä pysyy muistissa ja kaikki muu tippuu. Eli senkin sisälläs jos jotain pyydetään vain kerran niin seuraavat pyynnöt tiputtanee sen jossain kohtaa kun voimassaolevaa tavaraa kuitenkin kysellään vähän useammin. 

Sinällään tämän raiteen osoitepistecachen ei ehkä kannattaisi olla noin iso, sillä cachetettava data on aika iso per-raide. Kaikki osoitepisteet tarvitaan oikeastaan vain validoinnissa/julkaisussa/ratko-pushissa ja noissa käsitellään kerralla yhtä julkaisua eli muutamia raideversioita. Tässä voisi ehkä olla pienempi aika ja noinluokkaa 100 raidetta olisi varmaan jo aika riittävä. Toisaalta näitä cacheja voisi varmaan optimoida laajemminkin, checkaten että parametrit vastaa käyttöä :thinking:

Reply:
... ja heti perään itseni kiistäen: käytetäänköhän tätä cachea myös historiadatan plarailussa julkaisun muutoslistalla? Siellähän niitä raideversioita toki tulisi paljon enemmän. Toisaalta en ole ihan varma onko siellä isompi paha että lataus kestää vai se että syödään muistia. Ehkä tuollainen 2k pistelitaniaa ei kuitenkaan ole ihan gigatolkulla muistia.